eng/scripts/smoketest/Save-Package-Properties.ps1

```diff
@@ -0,0 +1,122 @@
+<#
+.SYNOPSIS
+Saves package properties from source into JSON files
+
+.DESCRIPTION
+Saves package properties in source of a given service directory to JSON files.
+JSON files are named in the form <package name>.json or <artifact name>.json if
+an artifact name property is available in the package properties.
+
+Can optionally add a dev version property which can be used logic for daily 
+builds.
+
+.PARAMETER serviceDirectory
+Service directory in which to search for packages
+
+.PARAMETER outDirectory
+Output location (generally a package artifact directory in DevOps) for JSON 
+files
+
+.PARAMETER addDevVersion
+Reads the version out of the source and adds a DevVersion property to the 
+package properties JSON file. If the package properties JSON file already 
+exists, read the Version property from the existing package properties JSON file
+and set that as the Version property for the new output. This has the effect of
+"adding" a DevVersion property to the file which could be different from the 
+Verison property in that file.
+#>
+
+[CmdletBinding()]
+Param (
+  [string] $serviceDirectory,
+
+  [string] $outDirectory,
+  [switch] $addDevVersion
+)
+
+. (Join-Path $PSScriptRoot ../../common/scripts/common.ps1)
+
+function SetOutput($outputPath, $incomingPackageSpec) {
+
+  # If there is an exsiting package info json file read that and set that as output object which gets properties updated here.
+  if (Test-Path $outputPath)
+  {
+    Write-Host "Found existing package info json."
+    $outputObject = ConvertFrom-Json (Get-Content $outputPath -Raw)
+  }
+  else
+  {
+    $outputObject = $incomingPackageSpec
+  }
+  
+
```

Comment:
https://gist.github.com/ckairen/f638231d6d146fef7b73f35d6ca7f7e0#smoke-test-project-building-pipeline-specification

Reply:
So the other parts have been dealt with, and I'll need your help on steps 2,3,6 specified under the "Smoke-Test-Project Building Pipeline Specification" section. The BuildSmokeTestProject is basically the file that does all the steps in the Smoke-Test-Project Building Pipeline Specification" section.

Reply:
the steps in BuildSmokeTestProject.yml presented there right now are just to let the language team devs understand how the PackageInfo can be accessed and see the structure of the pipeline's directory as some of them aren't super familiar with the eng sys pipelines

Reply:
Gotcha.  2 is the part that I'm looking for and was under the assumption was covered here.    We'll need to replace lines 13 - 27 for the below with the package references that you're generating.

Reply:
Once we've got that working, I can integrate into my code updates and we can test end-to-end.

Reply:
I've updated the [Update-Dependencies.ps1](https://github.com/Azure/azure-sdk-for-net/blob/main/common/SmokeTests/SmokeTest/Update-Dependencies.ps1) script to read in the JSON emitted here and rewrite the smoke test project references.

Reply:
@jsquire just to catch you up on some offline discussion that we had I think Update-Dependencies will end up calling `Get-AllPkgProperties` directly to find all the packages we need to add references to.

Reply:
So, we'd be calling to get the properties in real-time instead of reading the JSON files and would do the T2 filtering as part of that?   Seems reasonable to me.

Reply:
Correct you would call the in real time and do whatever filter that makes sense. If you just want to limit to track to then $package.IsNewSDK should do that.